File: src/main/字符串/t520/detectCapitalUse.go
package t520

func detectCapitalUse(word string) bool {
	upperLen := 0
	for i := 0; i < len(word); i++ {
		if checkIsUpper(string(word[i])) {
			upperLen++
		}
	}
	if len(word) == upperLen || (upperLen == 1 && checkIsUpper(string(word[0])) || upperLen == 0) {
		return true
	}
	return false
}
func checkIsUpper(c string) bool {
	if c >= "A" && c <= "Z" {
		return true
	}
	return false
}

File: src/main/字符串/t520/DetectCapitalUse.java
package 字符串.t520;

/**
 * @author: sotowang
 * @date: 2021/11/14 17:58
 */
public class DetectCapitalUse {
    private boolean checkIsUpper(char c) {
        if (c >= 'A' && c <= 'Z') {
            return true;
        }
        return false;
    }

    public boolean detectCapitalUse(String word) {
        char[] cs = word.toCharArray();
        int upperLen = 0;
        for (int i = 0; i < cs.length; i++) {
            if (checkIsUpper(cs[i])) {
                upperLen++;
            }
        }
        if (cs.length == upperLen || (upperLen == 1 && checkIsUpper(cs[0])) ||upperLen == 0) {
            return true;
        }
        return false;
    }
}
